tests/test_meshtal_analysis.py

```diff
@@ -70,8 +70,80 @@ def test_find_mesh_tally_num(self):
         data = ut.get_lines(path)
         
         self.assertEqual(meshtal_analysis.find_mesh_tally_numbers(data), {214: 4})
-            
+ 
+          
+class add_mesh_test(unittest.TestCase):
+     
+    def test_add_mesh_error(self):
+        mesh1_test = meshtal_analysis.meshtally()
+        mesh1_test.ctype="6col"
+        mesh2_test = meshtal_analysis.meshtally()
+        mesh2_test.ctype="6col"
+        
+        mesh1_test.x_bounds = (-9.0, -9.1)
+        mesh1_test.y_bounds = (-9.0, -9.1)
+        mesh1_test.z_bounds = (1.4, 1.5)
+        mesh2_test.x_bounds = (9.0, 9.1)
+        mesh2_test.y_bounds = (9.0, 9.1)
+        mesh2_test.z_bounds = (7.8, 7.9)
+        
+        with self.assertRaises(ValueError):
+            meshtal_analysis.add_mesh(mesh1_test, mesh2_test)
+        
+    def test_add_mesh(self):
+        mesh3_test = meshtal_analysis.meshtally()
+        mesh3_test.ctype="6col"
+        mesh4_test = meshtal_analysis.meshtally()
+        mesh4_test.ctype="6col"
+        
+        mesh3_test.data = [['1.000E+36', '-9.0', '-9.0', '1.4', '7.329430e-07', '0.017765']]
+        mesh4_test.data = [['1.000E+36', '-9.0', '-9.0', '1.4', '6.566330e-07', '0.018680']]
+        mesh3_test.data = meshtal_analysis.convert_to_df(mesh3_test)
+        mesh4_test.data = meshtal_analysis.convert_to_df(mesh4_test)
+        
+        mesh3_test.x_bounds = (-9, -9.1)
+        mesh3_test.y_bounds = (-9, -9.1)
+        mesh3_test.z_bounds = (1.4, 1.5)
+        mesh4_test.x_bounds = (-9, -9.1)
+        mesh4_test.y_bounds = (-9, -9.1)
+        mesh4_test.z_bounds = (1.4, 1.5)
+        
+        mesh3_test.data['value'] = 7.329430e-07
+        mesh3_test.data['rel_err'] = 0.017765
+        mesh4_test.data['value'] = 6.566330e-07
+        mesh4_test.data['rel_err'] = 0.018680
+        
+        new_mesh_test = meshtal_analysis.add_mesh(mesh3_test, mesh4_test)
+        
+        self.assertEqual(new_mesh_test.x_bounds, (-9, -9.1))
+        self.assertEqual(new_mesh_test.y_bounds, (-9, -9.1))
+        self.assertEqual(new_mesh_test.z_bounds, (1.4, 1.5))
+        self.assertEqual(new_mesh_test.data['value'].item(), 1.389576e-06)
+        self.assertEqual(new_mesh_test.data['rel_err'].item(),  0.025778627290839205)
+        
+         
+        
 
+class find_nearest_mid_test(unittest.TestCase):
+    
+    def test_find_nearest_mid(self):
+        test_val_1 = 4.6
+        test_mids_1 = [4, 5]
+        test_val_2 = 10
+        test_mids_2 = [2, 12, 11]
```

Comment:
this set of values should not be possible, the mids must be in order from lowest to higest